src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -426,6 +492,7 @@ def get_result_set(
                 self.run_query(
```

Comment:
hmm I just realized that plan.post_actions should be executed after the async query is done and should be executed here. We can discuss more tomorrow.